pkg/apis/pipeline/v1/pod.go

```diff
@@ -0,0 +1,112 @@
+/*
+Copyright 2022 The Tekton Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1
+
+import "github.com/tektoncd/pipeline/pkg/apis/pipeline/pod"
+
+// PodTemplate holds pod specific configuration
+type PodTemplate = pod.Template
+
+// MergePodTemplateWithDefault merges 2 PodTemplates together. If the same
+// field is set on both templates, the value from tpl will overwrite the value
+// from defaultTpl.
+func MergePodTemplateWithDefault(tpl, defaultTpl *PodTemplate) *PodTemplate {
```

Comment:
Can any of this be pulled out of the apis package?

Reply:
Can I ask for hint where do we put this? I tried to put under the pkg/pod package but that would introduce cyclic import.

Reply:
it can probably go in pkg/apis/pipeline/pod

Reply:
Thanks Lee. Added #5329